lib/vdsm/storage/mpathconf.py

```diff
@@ -432,6 +434,62 @@ def read_metadata():
     return Metadata(REVISION_MISSING, private)
 
 
+def check_local_config():
+    """
+    Check if 'user_friendly_names' is enabled in the multipathd configuration.
+    The configuration is obtained by running the multipathd command to show
+    the local configuration and capturing the output.
+    Parse the output and check that the contents are valid.
+
+    Raises:
+        exception.InvalidConfiguration: if any section have
+            'user_friendly_names' enabled. Contains the offending section name.
+    """
+    mpathconf = _parse_conf(io.StringIO(multipath.show_config_local()))
+    for conf in mpathconf:
+        failed_section, check_ok = _check_conf_attrs(conf.name, conf.attrs)
+        if not check_ok:
+            if failed_section is None:
+                failed_section = conf
+            raise exception.InvalidConfiguration(
+                reason=("'user_friendly_names' is enabled "
+                        "in multipath configuration"),
+                section=failed_section)
+        # Keep looping to ensure 'user_friendly_names' is not enabled by
+        # any other device.
+
+
+def _check_conf_attrs(name, attributes):
+    """
+    Check that 'user_friendly_names' is disabled in the attributes section
+    of the parsed multipathd configuration.
+    Search is recursive.
+
+    Args:
+        name (str): name of the section to print.
+        attributes: attributes in the configuration can contain
+            either nested sections or pairs.
+
+    Returns:
+        Tuple[Section, bool]:
+            Section: used if any nested section have 'user_friendly_names'
+                enabled. Contains the offending section tuple.
+            bool: all 'user_friendly_names' are disabled
+    """
+    for attr in attributes:
+        if isinstance(attr, Pair):
+            if attr.key == 'user_friendly_names' and attr.value == 'yes':
+                return None, False
+        else:
+            assert isinstance(attr, Section)
+            # Subsections (recursive call)
+            _, check_ok = _check_conf_attrs(
+                f"{name}.{attr.name}", attr.attrs)
+            if not check_ok:
+                return attr, check_ok
+    return None, True
```

Comment:
As a possibility, if all the children must have the same type, then we can use `all` function to avoid checking only the first child. It is less performant but here is not important, and with the data size that the multipath configuration outputs should not matter anyway.

Reply:
Then I can use `all` here, that will fix checking only the first child and ensure consistency in the sections.